op-service/eth/ssz.go

```diff
@@ -197,7 +211,8 @@ func (payload *ExecutionPayload) MarshalSSZ(w io.Writer) (n int, err error) {
 		offset += 4
 	}
 
-	if payload.inferVersion() == BlockV3 {
+	payloadVersion := payload.inferVersion()
+	if payloadVersion == BlockV3 || payloadVersion == BlockV4 {
```

Comment:
nit for easier maintenance:
```suggestion
	if payloadVersion.HasBlobProperties() {
```